include/rmm/device_scalar.hpp

```diff
@@ -37,6 +37,37 @@ class device_scalar {
  public:
   static_assert(std::is_trivially_copyable<T>::value, "Scalar type must be trivially copyable");
 
+  using value_type      = T;
+  using reference       = value_type &;
+  using const_reference = value_type const &;
+  using pointer         = value_type *;
+  using const_pointer   = value_type const *;
+  using iterator        = pointer;
+  using const_iterator  = const_pointer;
```

Comment:
I would define these in terms of `device_uvector`. Also, I don't think the `iterator` typedefs are needed for `device_scalar`? 
```suggestion
  using value_type      = typename device_uvector<T>::value_type;
  using reference       = typename device_uvector<T>::reference;
  using const_reference = typename device_uvector<T>::const_reference;
  using pointer         =  typename device_uvector<T>::pointer;
  using const_pointer   = typename device_uvector<T>::const_pointer;
```

Reply:
I had been thinking about whether it would be useful for device_scalar to provide `begin()` and `end()`...

Reply:
Done, thanks.